esp/logging/logginglib/logthread.cpp

```diff
@@ -414,3 +400,24 @@ IEspUpdateLogRequestWrap* CLogThread::unserializeLogRequestContent(const char* l
 
     return new CUpdateLogRequestWrap(guid, opt, buffer.str());
 };
+
+IEspUpdateLogRequestWrap* CLogThread::readWriteJobQueue(IEspUpdateLogRequestWrap* jobToWrite)
```

Comment:
I think extracting some of the code into a separate function is a good idea, but combining two different actions into the same function is not.  The only thing they share in common is that they need a critical section - they don't even share a parameter or a return value.  Much better would be "void writeJobQueue(item)" and IRequest * readJobQueue()

In this case a minimal change would probably be better, witht eh refactor as a separate commit.  I think it request the critical section to be moved into a local scope, something like:

```
   {
       CriticalBlock b(logQueueCrit);
       logRequest  = (IEspUpdateLogRequestWrap*)logQueue.dequeue();
   }

```